### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by GauravMakhijani**

### CODE REVIEW

File: db/website.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	ws = []WebsiteStatus{}
+	err = s.DB.Select(&ws, "SELECT id,link,status FROM links")
+
+	return
+}
+
+// InsertWebsite inserts a website into the database
+func (s *store) InsertWebsite(website WebsiteStatus) (err error) {
+
+	s.DB.Exec("INSERT INTO links (link, status) VALUES ($1, $2)", website.Link, website.Status)
```
Comment (sushmitaw) [inline db/website.go]: missing error handling

File: db/website.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	s.DB.Exec("UPDATE links SET status = $2 where link = $1", url, status)
+
+	return
+}
+
+// GetWebsiteStatus returns a particular website
+func (s *store) GetWebsiteStatus(query string) (ws []WebsiteStatus, err error) {
+
+	ws = []WebsiteStatus{}
+	err = s.DB.Select(&ws, "SELECT id,link,status FROM links WHERE link LIKE $1", "%"+query+"%")
```
Comment (sushmitaw) [inline db/website.go:44]: log the error

File: statuschecker/handler.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
+package statuschecker
+
+import (
+	"StatusChecker/db"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+func handlePostRequest(w http.ResponseWriter, r *http.Request, service Service) {
```
Comment (sushmitaw) [inline statuschecker/handler.go]: rename to createWebsiteHandler

File: statuschecker/handler.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+			continue
+		}
+	}
+
+	//send response
+	msg := "Websites added"
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(msg)
+
+	fmt.Println("Post request successful")
```
Comment (sushmitaw) [inline statuschecker/handler.go]: use log info instead of fmt

File: statuschecker/handler.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+func handleGetRequest(w http.ResponseWriter, r *http.Request, service Service) {
```
Comment (sushmitaw) [inline statuschecker/handler.go]: rename to getWebsiteHandler

File: statuschecker/handler.go [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+	//get query from url
+	query := r.URL.Query().Get("name")
```
Comment (sushmitaw) [inline statuschecker/handler.go]: query is wrong variable naming here it should be website or websiteName

File: statuschecker/handler.go [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(websites)
+
+	}
+
+	fmt.Println("Get request successful")
```
Comment (sushmitaw) [inline statuschecker/handler.go]: change all fmt to logs

File: statuschecker/handler.go [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+// HandleWebsites handles all requests to the /websites endpoint
+func HandleWebsites(service Service) func(w http.ResponseWriter, r *http.Request) {
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodPost { //only accept post requests
+
```
Comment (sushmitaw) [inline statuschecker/handler.go:82]: use go fmt ./... to format all code

File: statuschecker/service.go [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+func (s *statusService) GetSimilar(ctx context.Context, query string) (similar []db.WebsiteStatus, err error) {
+
+	//Make a call to the database to get the status of the website
+	similar, err = s.repo.GetWebsiteStatus(query)
+	return
+}
+
+func (s *statusService) Append(ctx context.Context, website db.WebsiteStatus) (err error) {
```
Comment (sushmitaw) [inline statuschecker/service.go]: Append is wrong naming convention use add or create

File: statuschecker/service.go [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+	s.repo.InsertWebsite(website)
```
Comment (sushmitaw) [inline statuschecker/service.go]: error handling missing

File: statuschecker/service.go [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+	//Make a call to the database to get the status of the website
+	// s.repo.GetWebsiteStatus(url)
```
Comment (sushmitaw) [inline statuschecker/service.go]: remove commented code if not needed

File: statuschecker/service.go [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+
+func (s *statusService) CheckStatus(ticker *time.Ticker) {
+	//return the status of the website
+
+	for {
+		select {
+		//wait for ticker to send a message
+		case <-ticker.C:
+
+			fmt.Println("Checking status of websites")
```
Comment (sushmitaw) [inline statuschecker/service.go]: use log

**Review by sushmitaw — CHANGES_REQUESTED** [review]

File: db/website.go (hunk 1 above)
Comment (GauravMakhijani) [reply db/website.go]: Done!

**Review by sushmitaw — APPROVED** [review]

**Pull request merged**